pkg/grafana/client/client.go

```diff
@@ -1,53 +1,105 @@
+	}
 
-	// Generate Grafana client
-	// Get grafana admin-password and admin-user
-	adminUserCredentials := AdminCredentials{
-		Username: conf.Environment.GrafanaAdminUsername,
-		Password: conf.Environment.GrafanaAdminPassword,
+	adminUsernameBytes, ok := adminSecret.Data[grafanaAdminSecretUserKey]
+	if !ok {
+		return nil, fmt.Errorf("key %q not found in Grafana admin secret %q", grafanaAdminSecretUserKey, grafanaAdminSecretName)
+	}
+	if len(adminUsernameBytes) == 0 {
+		return nil, fmt.Errorf("GrafanaAdminUsername is empty in secret %q", grafanaAdminSecretName)
 	}
-	if adminUserCredentials.Username == "" {
-		return nil, fmt.Errorf("GrafanaAdminUsername not set: %q", conf.Environment.GrafanaAdminUsername)
 
+	adminPasswordBytes, ok := adminSecret.Data[grafanaAdminSecretPasswordKey]
+	if !ok {
+		return nil, fmt.Errorf("key %q not found in Grafana admin secret %q", grafanaAdminSecretPasswordKey, grafanaAdminSecretName)
+	}
+	if len(adminPasswordBytes) == 0 {
+		return nil, fmt.Errorf("GrafanaAdminPassword is empty in secret %q", grafanaAdminSecretName)
 	}
-	if adminUserCredentials.Password == "" {
-		return nil, fmt.Errorf("GrafanaAdminPassword not set: %q", conf.Environment.GrafanaAdminPassword)
 
+	adminUserCredentials := AdminCredentials{
+		Username: string(adminUsernameBytes),
+		Password: string(adminPasswordBytes),
 	}
 
-	grafanaTLSConfig, err := TLSConfig{
-		Cert: conf.Environment.GrafanaTLSCertFile,
-		Key:  conf.Environment.GrafanaTLSKeyFile,
-	}.toTLSConfig()
+	// Get Grafana TLS configuration from secret, if it exists
+	var clientTLSConfig *tls.Config
+	tlsSecret := &corev1.Secret{}
+	tlsSecretKey := client.ObjectKey{Namespace: grafanaNamespace, Name: grafanaTLSSecretName}
+	err := k8sClient.Get(ctx, tlsSecretKey, tlsSecret)
 	if err != nil {
-		return nil, fmt.Errorf("failed to build tls config: %w", err)
+		return nil, fmt.Errorf("failed to get Grafana TLS secret %q in namespace %q: %w", grafanaTLSSecretName, grafanaNamespace, err)
+	}
+
+	// TLS Secret found, try to load cert and key
+	tlsCertBytes, certOk := tlsSecret.Data[grafanaTLSSecretCertKey]
+	tlsKeyBytes, keyOk := tlsSecret.Data[grafanaTLSSecretKeyKey]
+
+	if !certOk || len(tlsCertBytes) == 0 {
+		return nil, fmt.Errorf("key %q not found or empty in Grafana TLS secret %q", grafanaTLSSecretCertKey, grafanaTLSSecretName)
+	}
+	if !keyOk || len(tlsKeyBytes) == 0 {
+		return nil, fmt.Errorf("key %q not found or empty in Grafana TLS secret %q", grafanaTLSSecretKeyKey, grafanaTLSSecretName)
+	}
+
+	// tlsConfigInput is our local struct type from ./tls.go
+	tlsConfigInput := TLSConfig{
+		Cert: string(tlsCertBytes),
+		Key:  string(tlsKeyBytes),
+	}
+	// clientTLSConfig is the *tls.Config for the Grafana client
+	var buildErr error
+	clientTLSConfig, buildErr = tlsConfigInput.toTLSConfig()
+	if buildErr != nil {
+		return nil, fmt.Errorf("failed to build TLS config from secret %q: %w", grafanaTLSSecretName, buildErr)
```

Comment:
Can also just use `err` here

```suggestion
	clientTLSConfig, err = tlsConfigInput.toTLSConfig()
	if err != nil {
		return nil, fmt.Errorf("failed to build TLS config from secret %q: %w", grafanaTLSSecretName, err)
```